src/amo/components/PermissionsCard/index.js

```diff
@@ -0,0 +1,280 @@
+      find: i18n.gettext('Read the text of all open tabs'),
+      geolocation: i18n.gettext('Access your location'),
+      history: i18n.gettext('Access browsing history'),
+      management: i18n.gettext('Monitor extension usage and manage themes'),
+      // In Firefox the following message replaces the name "Firefox" with the
+      // current brand name, e.g., "Nightly", but we do not need to do that.
+      nativeMessaging: i18n.gettext('Exchange messages with programs other than Firefox'),
+      notifications: i18n.gettext('Display notifications to you'),
+      pkcs11: i18n.gettext('Provide cryptographic authentication services'),
+      proxy: i18n.gettext('Control browser proxy setting'),
+      privacy: i18n.gettext('Read and modify privacy settings'),
+      sessions: i18n.gettext('Access recently closed tabs'),
+      tabs: i18n.gettext('Access browser tabs'),
+      tabHide: i18n.gettext('Hide and show browser tabs'),
+      topSites: i18n.gettext('Access browsing history'),
+      unlimitedStorage: i18n.gettext('Store unlimited amount of client-side data'),
+      webNavigation: i18n.gettext('Access browser activity during navigation'),
+      allUrls: i18n.gettext('Access your data for all websites'),
+      wildcard: (domain) => {
+        return i18n.sprintf(
+          i18n.gettext('Access your data for sites in the %(domain)s domain'),
+          { domain }
+        );
+      },
+      tooManyWildcards: (count) => {
+        return i18n.sprintf(
+          i18n.gettext('Access your data in %(count)s other domains'),
+          { count: i18n.formatNumber(count) },
+        );
+      },
+      oneSite: (site) => {
+        return i18n.sprintf(
+          i18n.gettext('Access your data for %(site)s'),
+          { site }
+        );
+      },
+      tooManySites: (count) => {
+        return i18n.sprintf(
+          i18n.gettext('Access your data on %(count)s other sites'),
+          { count: i18n.formatNumber(count) },
+        );
+      },
+    };
+
+    // Classify a permission as a host permission or a regular permission.
+    function classifyPermission(permission) {
+      const match = /^(\w+)(?:\.(\w+)(?:\.\w+)*)?$/.exec(permission);
+      let result = { type: 'permissions', value: permission };
+      if (!match) {
+        result = { type: 'hosts', value: permission };
+      }
+      return result;
+    }
+
+    const permissionsToDisplay = [];
+
+    // Format and sequence all the Permission components.
+    function formatPermissions(addonPermissions) {
+      const permissions = { hosts: [], permissions: [] };
+
+      // First, categorize them into host permissions and regular permissions.
+      // eslint-disable-next-line no-restricted-syntax
+      for (const permission of addonPermissions) {
+        const { type, value } = classifyPermission(permission);
+        permissions[type].push(value);
+      }
+
+      // Classify the host permissions.
```

Comment:
This code could also be extracted out into a function in a helper to make it more testable. As it is I think I've covered most, if not all of it, in the tests for `PermissionsCard`, so maybe it's not that important.